### PR TITLE
Add customizable Bot status

### LINKED ISSUE #569: Enhancement - bot status
Add a bot status like listening to /play or something to explain users that dont have a brain.

### PR DESCRIPTION
**Pull request by Zagrthos**
Closes #569 

- Added presence before the bot is fully initialized
- Added new Variables `BOT_STATUS`, `BOT_ACTIVITY_TYPE`, `BOT_ACTIVITY_URL` & `BOT_ACTIVITY` to .env Configuration File (also updated the .example)
- Updated README to show new possibilities

You can read all Activity Types [here](https://discord.js.org/#/docs/discord.js/stable/typedef/ActivityType) and all status [here](https://discord.js.org/#/docs/discord.js/stable/typedef/PresenceStatusData)

- [X] I updated the changelog

Made the PR a second time because GitHub wanted to merge the other branch to this....

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -6,3 +6,6 @@ SPOTIFY_CLIENT_SECRET=
 
 # Optional
 # CACHE_LIMIT=2GB
+# BOT_STATUS=
```
Comment (codetheweb) [inline .env.example:9]: Could you add a comment with a link to the possible values for these variables?

A section should also be added to the README about them.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- Added a configurable Bot Status with user defined Activities
```
Comment (codetheweb) [inline CHANGELOG.md:9]:
```suggestion
- Added a configurable bot status with user defined activities
```

File: src/bot.ts [hunk 3]
```diff
@@ -33,6 +36,9 @@ export default class {
     this.shouldRegisterCommandsOnBot = config.REGISTER_COMMANDS_ON_BOT;
     this.commandsByName = new Collection();
     this.commandsByButtonId = new Collection();
+    this.cbot_status = config.BOT_STATUS;
```
Comment (codetheweb) [inline src/bot.ts:39]: Let's add a new `.config` property to this class so we don't have to extract all of these out.

**Review by codetheweb — CHANGES_REQUESTED** [review]

File: .env.example (hunk 1 above)
Comment (Zagrthos) [reply .env.example:11]: Both done :)

File: src/bot.ts (hunk 3 above)
Comment (Zagrthos) [reply src/bot.ts:39]: bot.ts
```
constructor(
  @inject(TYPES.Client) client: Client,
    @inject(TYPES.Config) config: Config,
      @inject(TYPES.ConfigStatus) configStatus: Config,
  ) {
    this.client = client;
    this.token = config.DISCORD_TOKEN;
    this.shouldRegisterCommandsOnBot = config.REGISTER_COMMANDS_ON_BOT;
    this.commandsByName = new Collection();
    this.commandsByButtonId = new Collection();
    this.cbot_status = configStatus.BOT_STATUS;
    this.cbot_act_type = configStatus.BOT_ACTIVITY_TYPE;
    this.cbot_act = configStatus.BOT_ACTIVITY;
  }
```

inversify.config.ts
```
// Config values
container.bind(TYPES.Config).toConstantValue(new ConfigProvider());
container.bind(TYPES.ConfigStatus).toConstantValue(new ConfigProvider());

```

types.ts
```ConfigStatus: Symbol('ConfigStatus'),````

Like this?

Comment (Zagrthos) [reply src/bot.ts]: Changed it in the newest [commit](https://github.com/codetheweb/muse/pull/599/commits/e3b1d61ddd1328db29ee1b1ef7d2fbaa49dfeb99). Seems to work fine and looks better.

**Comment by codetheweb** [comment]
Should be good to go after those type issues are fixed.

**Comment by Zagrthos** [comment]
Added the last Activity (Streaming). If you're good with it I'm finished. :)

**Pull request merged**